src/Microsoft.VisualStudio.Extensibility.Testing.Xunit.Shared/Harness/VisualStudioInstanceFactory.cs

```diff
@@ -414,7 +414,20 @@ private static async Task<Process> StartNewVisualStudioProcessAsync(string insta
 
             ProcessStartInfo CreateSilentStartInfo(string fileName, string arguments)
             {
-                return new ProcessStartInfo(fileName, arguments) { CreateNoWindow = true, UseShellExecute = false };
+                var startInfo = new ProcessStartInfo(fileName, arguments) { CreateNoWindow = true, UseShellExecute = false };
+                foreach (var variable in environmentVariables)
+                {
+                    if (string.IsNullOrEmpty(variable.Value))
+                    {
+                        startInfo.EnvironmentVariables.Remove(variable.Key);
```

Comment:
is the presence of an env variable different from it having an empty value?  Wondering if we could always take the else path instead

Reply:
I'm not certain. I'm using this based on what I see in other projects.